resources/views/external-link-confirm.blade.php

```diff
@@ -6,61 +6,86 @@
 
 @php($identifier = md5($url . '-' . Str::random(8)))
 
-<a
-    href="javascript:;"
-    x-data="{
+<div x-data="{
         openModal() {
             Livewire.emit('openModal', '{{ $identifier }}')
+        },
+        redirect() {
+            window.open('{{ $url }}', '_blank')
+        },
+        hasDisabledLinkWarning() {
+            return localStorage.getItem('has_disabled_link_warning') === 'true';
         }
     }"
-    class="inline-flex items-center space-x-2 font-semibold whitespace-nowrap cursor-pointer link"
-    @click="openModal"
 >
-    <span>{{ $text ?? $slot ?? '' }}</span>
+    <a
+        :href="hasDisabledLinkWarning() ? '{{ $url }}' : 'javascript:;'"
+        :target="hasDisabledLinkWarning() ? '_blank' : '_self'"
+        class="inline-flex items-center space-x-2 font-semibold whitespace-nowrap cursor-pointer link"
+        @click="hasDisabledLinkWarning() ? redirect() : openModal()"
+    >
+        <span>{{ $text ?? $slot ?? '' }}</span>
 
-    @unless($noIcon)
-        <x-ark-icon name="link" size="sm" class="mx-2" />
-    @endunless
-</a>
+        @unless($noIcon)
+            <x-ark-icon name="link" size="sm" class="mx-2" />
+        @endunless
+    </a>
 
-@push('footer')
-    <x-ark-js-modal
-        :name="$identifier"
-        class="w-full max-w-2xl text-left rounded-xl"
-        title-class="header-2"
-        buttons-style="flex justify-end space-x-3"
-        init
-    >
-        @slot('title')
-            @lang('generic.external_link')
-        @endslot
+    @push('footer')
+        <x-ark-js-modal
+            :name="$identifier"
+            class="w-full max-w-2xl text-left rounded-xl"
+            title-class="header-2"
+            buttons-style="flex justify-end space-x-3"
+            x-data="{
+                hasConfirmedLinkWarning: false,
```

Comment:
this is still stored between opening, so if I click the checkbox and then close and reopen the modal it's still checked.